src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeBuffer.cs

```diff
@@ -293,25 +300,32 @@ public void WriteArray<T>(ulong byteOffset, T[] array, int index, int count)
             if (array.Length - index < count)
                 throw new ArgumentException(SR.Argument_InvalidOffLen);
 
+            WriteSpan(byteOffset, new ReadOnlySpan<T>(array, index, count));
+        }
+
+        [CLSCompliant(false)]
+        public void WriteSpan<T>(ulong byteOffset, ReadOnlySpan<T> data)
+            where T : struct
+        {
             if (_numBytes == Uninitialized)
                 throw NotInitialized();
 
             uint sizeofT = SizeOf<T>();
             uint alignedSizeofT = AlignedSizeOf<T>();
             byte* ptr = (byte*)handle + byteOffset;
-            SpaceCheck(ptr, checked((nuint)(alignedSizeofT * count)));
+            SpaceCheck(ptr, checked((nuint)(alignedSizeofT * data.Length)));
 
             bool mustCallRelease = false;
             try
             {
                 DangerousAddRef(ref mustCallRelease);
 
-                if (count > 0)
+                if (!data.IsEmpty)
```

Comment:
This condition is not needed (it was needed in the original code to prevent `array[index]` from throwing).

Reply:
I think it can be checked earlier to reduce the overhead of check and pinning. Is copying to zero-lengthed span always valid and being a nop, if the SafeBuffer is initialized?

Reply:
I do not think it is interesting to have extra code to optimize this method for empty input. People generally do not call APIs like this with empty input.